wagtail/images/templatetags/wagtailimages_tags.py

```diff
@@ -200,22 +200,32 @@ def render(self, context):
         if not image:
             return ""
 
+        # Get original image dimensions
+        original_width, original_height = image.width, image.height
+
+        # Get renditions and filter them
         renditions = get_renditions_or_not_found(
             image,
             self.get_filters(preserve_svg=self.preserve_svg and image.is_svg()),
         )
 
+        # Filter out renditions larger than the original image size
+        filtered_renditions = [
```

Comment:
Suggestion: It would be more efficient to filter out the larger renditions before the `get_renditions_or_not_found` call, since it would save the need to fetch them from the DB, or perhaps create them in the first place.

Reply:
Filtering the renditions before calling get_renditions_or_not_found would indeed save us from some unnecessary database fetches and potential processing. I'll make the change to filter the size before trying to fetch renditions so we don't attempt to fetch any renditions larger than the original image. Thanks for the suggestion!

Reply:
I think the problem there is that the "calculate what size this rendition should be" logic is currently coupled to the actual rendition creation, so there's no way to know which ones should be filtered out without having the rendition objects.

Reply:
the rendition creation procedure actually computes the dimensions at runtime, which makes filtering out sizes before creation kind of tricky. Maybe it might be decoupling of size checking logic from a creation step, and with that decoupling filtering could occur without the use of full rendition objects maybe. Maybe this is also something involving a helper class that does some estimate to dimensions based on specs; thus helping catch oversized renditions early. Would you think this is a realistic approach?